executor/server.go

```diff
@@ -106,42 +107,46 @@ func (s *Server) Start(ctx context.Context) error {
 		Capability: 100,
 	}
 
-	resp, err := s.cli.RegisterExecutor(ctx, registerReq)
+	resp, err := s.cli.RegisterExecutor(ctx, registerReq, s.cfg.RPCTimeout)
 	if err != nil {
 		return err
 	}
 	s.ID = model.ExecutorID(resp.ExecutorId)
 	log.L().Logger.Info("register successful", zap.Int32("id", int32(s.ID)))
 
 	// Start Heartbeat
-	ticker := time.NewTicker(time.Duration(s.cfg.KeepAliveInterval) * time.Millisecond)
+	ticker := time.NewTicker(s.cfg.KeepAliveInterval)
+	s.lastHearbeatTime = time.Now()
 	for {
 		select {
 		case <-ctx.Done():
 			return nil
 		case <-exitCh:
 			return nil
 		case t := <-ticker.C:
```

Comment:
In L148 `s.lastHearbeatTime` is set to t, is it better to set `s.lastHearbeatTime = time.Now()`, since heartbeat rpc could cost some time.

Reply:
We aim to keep `lastHbTime of executor` consistant with `lastHbTime of Master`. If we set the heartbeat time of executor to the `start time` of rpc, it will a little bit ealier than the heartbeat time of master, which is safe. In contrast, if we set it to the `end time` of rpc, it might be a little bit later than master's, which might cause that master wait for less time than executor. This gap is unsafe.